### PR TITLE
chore(types): lint types

### PR DESCRIPTION
**Pull request by TheAlexLichter**
This PR provides linting support for Nuxt's typescript types.

As my TS skills are not the best, I'd love to have some eyes on the PR 😋

## Types of changes
- [x] Internal setup improvement

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [x] I have added tests to cover my changes (if not applicable, please state why)
- [x] All new and existing tests are passing.

### CODE REVIEW

File: packages/types/config/index.d.ts [hunk 1]
```diff
@@ -25,7 +25,7 @@ export { ServerMiddleware } from './server-middleware'
 /**
  * @deprecated Use NuxtConfig instead
 */
-export interface Configuration extends Record<string, any> {}
+export type Configuration = Record<string, any>
```
Comment (pi0) [inline packages/types/config/index.d.ts:28]: Using interface was intentional i guess to be able extending. /cc @kevinmarrec

Comment (manniL) [reply packages/types/config/index.d.ts:28]: Makes sense! Reveting back here 😋

**Review by pi0 — APPROVED** [review]

**Review by mary-doc — COMMENTED** [review]
Thank you for sharing your thoughts, comments, and ideas with us! We read and evaluate all feedback carefully, but we may not be able to respond to every submission.

**Review by mary-doc — COMMENTED** [review]
Thank you for sharing your thoughts, comments, and ideas with us! We read and evaluate all feedback carefully, but we may not be able to respond to every submission.

**Review by mary-doc — COMMENTED** [review]
Submit general feedback with explicit approval

**Pull request merged**